### ISSUE TITLE
Rewrite doesn't switch trig to hyperbolic functions

### ISSUE THREAD

**Issue by GodOfDeathSam** [sympy/sympy#24199]
Running the following code:
```
coth(x).rewrite(cot)
```
Output is
![image](https://user-images.githubusercontent.com/87801374/198899278-1e66ca3f-de66-43e2-970a-4bdb2d91644f.png)

Expect to get 
![image](https://user-images.githubusercontent.com/87801374/198899460-a0e3ec9f-92f1-4ee9-8375-f81a4a369008.png)

**Comment by oscarbenjamin** [comment]
I guess there are many of these:
```python
In [1]: coth(x).rewrite(cot)
Out[1]: coth(x)

In [2]: cosh(x).rewrite(cos)
Out[2]: cosh(x)

In [3]: tanh(x).rewrite(tan)
Out[3]: tanh(x)

In [4]: cot(x).rewrite(coth)
Out[4]: cot(x)
```
It shouldn't be too hard to add the methods like `_eval_rewrite_as_cot` etc.

**Comment by ramvikrams** [comment]
Can I take it up

**Comment by oscarbenjamin** [comment]
Anyone is welcome to work on anything.

**Comment by ramvikrams** [comment]
thanks

**Comment by ramvikrams** [comment]
```
def _eval_rewrite_as_cos(self, arg, **kwargs):
      return Icos(I * arg)
```
Can this be a possible fix for ```cosh(x).rewrite(cos)```

**Comment by asmeurer** [comment]
This can also be an opportunity to update some of the code to use the new rewrite API (see https://docs.sympy.org/latest/guides/custom-functions.html#rewrite).

**Comment by sylee957** [comment]
Even if you implement the rewrite, The rewrite won't take any effect anyway because of the the automatic evaluation.
I already see that `cos(I*arg)` automatically becomes `cosh`

**Comment by ramvikrams** [comment]
> Even if you implement the rewrite, The rewrite won't take any effect anyway because of the the automatic evaluation.
> I already see that `cos(I*arg)` automatically becomes `cosh`

So how to change the automatic evaluation

**Comment by GodOfDeathSam** [comment]
I guess smth like
```
return I*cos(I*arg, evaluate=False)
```

**Comment by ramvikrams** [comment]
> I guess smth like
> 
> ```
> return I*cos(I*arg, evaluate=False)
> ```

oh thanks

**Comment by oscarbenjamin** [comment]
I'm not sure that we should create unevaluated expressions by default like that.

**Comment by ramvikrams** [comment]
what can we do then

**Comment by asmeurer** [comment]
We should change the trig functions to not evaluate so much. Ideally they should only evaluate on numeric arguments and evaluation on symbolic arguments like `I*x` should be moved to `doit()` (see https://docs.sympy.org/latest/guides/custom-functions.html#best-practices-for-eval). However, this may not be straightforward change, as some things might be relying on this evaluation.

**Comment by oscarbenjamin** [comment]
We can also add the rewrite functions that do work anyway. There are many possible combinations here.

**Comment by ramvikrams** [comment]
Thanks sir for help

**Comment by ramvikrams** [comment]
I read the sympy documentation for doit(),eval(). So there are few things which I wan't to discuss.

+ In the eval funtion created in the hyperbolic function, should we do the return None for certain arguments which we need don't need to evaluate and then the problem will be solved
+ Or a doit() can be created 
+ Or we can create a custom function for the arguments in the hyperbolic funtion

Please can you sujjest with which method it is best to proceed

**Comment by asmeurer** [comment]
The main thing is to remove it from eval. It may be useful to move the expansion to another method so it can still be accessed, like `doit` or maybe just `_eval_expand_trig`.

**Comment by ramvikrams** [comment]
> The main thing is to remove it from eval. It may be useful to move the expansion to another method so it can still be accessed, like `doit` or maybe just `_eval_expand_trig`.

can we do the return type None there in eval function

**Comment by asmeurer** [comment]
Yes, eval should return None when the function should remain unevaluated. The first thing to do would be to try changing this behavior then running the tests to see what fails. There may things that are relying on this behavior which would have to be fixed in order to change it.

**Comment by ramvikrams** [comment]
> avior then running the tests to see what fa

Thanks sir I'll get started with it

**Comment by harshkasat** [comment]
> The main thing is to remove it from eval. It may be useful to move the expansion to another method so it can still be accessed, like `doit` or maybe just `_eval_expand_trig`.

@asmeurer It seems like we need to create a new method called `eval_expand_trig` to evaluate `False`, as mentioned in the PR and above conversation. Is this correct? @ramvikrams  also mention doit() also evaluate unevaluated expr( [doit](https://docs.sympy.org/latest/guides/custom-functions.html#custom-functions-doit) )
